test/backward_compatibility/check_backward_compatibility.py

```diff
@@ -25,6 +25,17 @@
     ('aten::real*', datetime.date(2020, 4, 15)),
     ('aten::imag*', datetime.date(2020, 4, 15)),
     ('aten::quantize_per_tensor', datetime.date(2020, 4, 15)),
+    ('quantized::conv_prepack', datetime.date(2020, 4, 15)),
+    ('quantized::conv_unpack', datetime.date(2020, 4, 15)),
+    ('quantized::conv', datetime.date(2020, 4, 15)),
+    ('quantized::conv2d_prepack', datetime.date(2020, 4, 15)),
+    ('quantized::conv2d_unpack', datetime.date(2020, 4, 15)),
+    ('quantized::conv2d', datetime.date(2020, 4, 15)),
+    ('quantized::conv2d_relu', datetime.date(2020, 4, 15)),
+    ('quantized::conv3d_prepack', datetime.date(2020, 4, 15)),
+    ('quantized::conv3d_unpack', datetime.date(2020, 4, 15)),
+    ('quantized::conv3d', datetime.date(2020, 4, 15)),
+    ('quantized::conv3d_relu', datetime.date(2020, 4, 15)),
```

Comment:
I think we're also going to have to figure out if this will break serialized TorchScript modules. The way that would happen is if the packed parameter attributes in the serialized Module are typed as Tensor, rather than Conv*dPackedParamsBase. Can you jit.save() a model containing one of these modules before this change, load it in after this change, and confirm if this is indeed a breaking situation?

Reply:
is `torch.jit.save` calling `__getstate__` and `__setstate__` function? I didn't change these functions so it should still work(we are still saving unpacked weights directly)

Reply:
Yes, this will break serialized torchscript modules since the whole module is saved with `torch.jit.save`:
```
import torch
class M(torch.nn.Module):
    def __init__(self):
        super(M, self).__init__()
        self.conv = torch.nn.quantized.Conv2d(3, 3, 3)

    def forward(self, x):
        return self.conv(x)

a = torch.jit.script(M())
x = torch.quantize_per_tensor(torch.rand(1, 3, 10, 10), 0.5, 5, torch.quint8)
b = torch.jit.trace(M(), x)
print(a)
print(b)
# torch.jit.save(a, "a.pt")
# torch.jit.save(b, "b.pt")

a = torch.jit.load("a.pt")
b = torch.jit.load("b.pt")
print(a, b)
```

Error:
```
Traceback (most recent call last):
  File "c.py", line 18, in <module>
    a = torch.jit.load("a.pt")
  File "/data/users/jerryzh/anaconda3/envs/py3/lib/python3.7/site-packages/torch/jit/__init__.py", line 227, in load
    cpp_module = torch._C.import_ir_module(cu, f, map_location, _extra_files)
RuntimeError:

quantized::conv2d(Tensor _0, __torch__.torch.classes.quantized.Conv2dPackedParamsBase _1, float _2, int _3) -> (Tensor _0):
Expected a value of type '__torch__.torch.classes.quantized.Conv2dPackedParamsBase' for argument '_1' but instead found type 'Tensor'.
:
  File "/data/users/jerryzh/anaconda3/envs/py3/lib/python3.7/site-packages/torch/nn/quantized/modules/conv.py", line 337
        if len(input.shape) != 4:
            raise ValueError("Input shape must be `(N, C, H, W)`!")
        return ops.quantized.conv2d(
               ~~~~~~~~~~~~~~~~~~~~ <--- HERE
            input, self._packed_params, self.stride, self.padding,
            self.dilation, self.groups, self.scale, self.zero_point)
Serialized   File "code/__torch__/torch/nn/quantized/modules/conv.py", line 36
    _13 = [_11, _12]
    _14, _15, = _4
    _16 = ops.quantized.conv2d(input, _1, _10, _13, [_14, _15], _5, _6, _7)
          ~~~~~~~~~~~~~~~~~~~~ <--- HERE
    return _16
  def __getstate__(self: __torch__.torch.nn.quantized.modules.conv.Conv2d) -> Tuple[int, int, Tuple[int, int], Tuple[int, int], Tuple[int,
int], Tuple[int, int], bool, int, int, str, Tensor, Optional[Tensor], float, int, bool]:
```

Reply:
@jerryzh168 right, the issue here is that the ScriptModules are serialized with the packed parameters annotated as type Tensor. You can probably fix this by replicating what I've done on the import path: https://github.com/pytorch/pytorch/pull/34140/files#diff-93507b8ace0ea4890fdafb4f18e55f46R265

Reply:
BC is fixed now for `torch.jit.save`

Reply:
For documentation purpose, here is what we do(paste from chat): 
1) two overloads of conv2d, one with the old schema and one with the new. the one with the old schema throws an exception and says to use the new one
2) a pass that converts the old conv2d callsites to the new schema on ScriptModules after load()